packages/@expo/cli/src/start/server/metro/withMetroMultiPlatform.ts

```diff
@@ -297,11 +297,14 @@ export function withExtendedResolver(
     };
   }
 
+  // TODO: This is a hack to get resolveWeak working.
+  const idFactory = config.serializer?.createModuleIdFactory?.() ?? ((id: number): number => id);
```

Comment:
I think we also need to account for `string` support, right? As that's added in the newer versions of Metro.

```suggestion
  const idFactory = config.serializer?.createModuleIdFactory?.() ?? ((id: number | string): number | string => id);
```